src/components/Home/Home.jsx

```diff
@@ -0,0 +1,235 @@
+      .then((result) => {
+        const newData = data.map((item) => {
+          if (item._id === result._id) {
+            return result;
+          } else {
+            return item;
+          }
+        });
+
+        setData(newData);
+      })
+      .catch((err) => console.log(err));
+  };
+
+  const deletePost = (postId) => {
+    fetch(`https://chit-chat-12.herokuapp.com/delete/${postId}`, {
+      method: "delete",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: "Bearer " + localStorage.getItem("jwt"),
+      },
+    })
+      .then((res) => res.json())
+      .then((result) => {
+        const newData = data.filter((elem) => {
+          return elem._id !== result.result._id;
+        });
+
+        setData(newData);
+      })
+      .catch((err) => console.log(err));
+  };
+
+  // console.log(data)
+
+  const temp = JSON.parse(localStorage.getItem("user"));
+
+  return (
+    <div className="body">
+      <Link to="/createpost">
+        <h2 className="create-post">
+          <AddCircleIcon fontSize="large" />{" "}
+        </h2>
+      </Link>
+      <div className="media_content">
+        {/* Card */}
+
+        {data &&
+          data.map((elem) => {
+            return (
+              <div className="media-card">
+                <div>
+                  <div
+                    style={{
+                      backgroundColor: "white",
+                      display: "flex",
+                      justifyContent: "space-between",
+                    }}
+                  >
+                    <h5
+                      style={{ paddingLeft: "0.5rem", paddingBottom: "1rem" }}
+                    >
+                      {" "}
+                      <Link
+                        to={
+                          elem.postedBy._id !== state._id
+                            ? "/profile/" + elem.postedBy._id
+                            : "/profile"
+                        }
+                      >
+                        {elem.postedBy.name}
+                      </Link>{" "}
+                    </h5>
+                    <IconButton>
+                      <DeleteIcon
+                        style={{
+                          dispaly:
+                            temp._id !== elem.postedBy._id ? "none" : "block",
+                        }}
+                        onClick={() => deletePost(elem._id)}
+                      />
+                    </IconButton>
+                  </div>
+                  <img src={elem.photo} alt="" />
+                  <h3 style={{ paddingLeft: "0.5rem" }}>{elem.title}</h3>
+
+                  {elem.likes.includes(state._id) ? (
+                    <IconButton>
+                      <ThumbDownAltIcon onClick={() => postUnlikes(elem._id)} />
+                    </IconButton>
+                  ) : (
+                    <IconButton>
+                      <ThumbUpIcon onClick={() => postLikes(elem._id)} />
```

Comment:
I can increase the number of likes by more than 1 by rapidly clicking over the like button. Ideally I should be able to increase likes by 1 only.